core/unit_test/CMakeLists.txt

```diff
@@ -293,5 +294,61 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   SOURCES UnitTestMain.cpp  TestHostBarrier.cpp
 )
 
+FUNCTION (TEST_FEATURE_IMPLEMENTED CURRENT_FEATURE FEATURE_LIST)
+  SET ( FEATURE_IMPLEMENTED OFF PARENT_SCOPE )
+  FOREACH ( IMPLEMENTED_FEATURE ${FEATURE_LIST})
+    message( STATUS "OUT:${IMPLEMENTED_FEATURE}, ${CURRENT_FEATURE}" )
+    if(CURRENT_FEATURE STREQUAL ${IMPLEMENTED_FEATURE} )
+      set (FEATURE_IMPLEMENTED ON PARENT_SCOPE )
+    endif()
+  ENDFOREACH()
+ENDFUNCTION()
+
+
+FUNCTION (KOKKOS_ADD_INCREMENTAL_TEST BACK_END_NAME)
+
+  # Iterate over incremental tests in directory
+    # Call here the bash script to get the file list in the test directory
+    # OUTPUT: List of test file names
+  SET( INCREMENTAL_FILE_LIST ExecSpaceCompilation.hpp )
+  SET( ${BACK_END_NAME}_SOURCES UnitTestMainInit.cpp )
+
+  foreach (CURRENT_FILE_NAME ${INCREMENTAL_FILE_LIST})
+    string (REPLACE ".hpp" "" CURRENT_TEST_NAME ${CURRENT_FILE_NAME})
+
+    message( STATUS "OUT:${CURRENT_TEST_NAME}" )
+    # From each test get list of dependencies 
+      # Run a bash script to get the list of dependencies for this test
+      # OUTPUT: List of Kokkos features required to run the test
+    SET( CURRENT_REQUIRED_FEATURE_LIST "ExecSpaceDef" )
```

Comment:
note that this is a place holder...variable CURRENT_REQUIRED_FEATURE_LIST will be populated with script above

Reply:
I have the feeling while potentially powerful, having a full dependency list of features is more than we need. I would be fairly comfortable with just numbering the tests through, and have for each execution space a max number up to which stuff is implemented, and maybe a list of "to-be-skipped" tests.

Reply:
@Rombur said the move to a feature set was necessary, that there isn't a strict ordering of tests. Bruno, would a capability number plus a list of tests to skip meet your needs?

Reply:
Yes, that would work.

Reply:
updated this to use a required level and the -DKokkos_<>_FEATURE_LEVEL must be greater than or equal to required level in order for the test to be included.

Reply:
Also added test exclusion logic

Reply:
This looks much better to me. Thanks.